internal/provider/kubernetes/routes.go

```diff
@@ -497,7 +586,26 @@ func (r *gatewayAPIReconciler) processUDPRoutes(ctx context.Context, gatewayName
 		return err
 	}
 
-	for _, udpRoute := range udpRouteList.Items {
+	udpRoutes := udpRouteList.Items
```

Comment:
can all this code be consolidated by using a helper func with a client.Object signature ?

Reply:
ahhh. Good catch. I didn't realise that `client.Object` can be used!

Reply:
Unfortunately, `AuthenticaionFilter`, RateLimitFilter` and `unstructured.Unstructured` don't implment the common interface. The alternative I could do is to add a NamespaceGetter interface to simplify the code.

Reply:
you could use existing code for them, and add a `TODO`